intelligent-video-recording/gst_display.py

```diff
@@ -104,31 +109,58 @@ def parseLabels(self, file):
+        # generates a signal parsed as a JSON 
+        # See: 
+        # https://developer.ridgerun.com/wiki/index.php?title=GstInference/Metadatas/GstInferenceMeta
+
+        # Detection models use predictions category from JSON meta
+        predictions_list = data["predictions"]
+        for item in predictions_list:
+            # Parse class id from prediction
+            classes.append(item["classes"][0]["Class"])
+            # Parse probability from prediction. Handle ',' float notation.
+            probabilities.append(float(item["classes"][0]["Probability"].replace(",",".")))
+
+        # Classification models use classes category from JSON meta
+        predictions_list_classes = data["classes"]
+        for item in predictions_list_classes:
+            # Parse class id from prediction
+            class_id = item["Class"]
+            # Parse probability from prediction. Handle ',' float notation.
+            class_probability = float(item["Probability"].replace(",","."))
+
+            classes.append(class_id)
+            probabilities.append(class_probability)
 
         # Detect require class ID and min probability threshold
-        if(class_id in self.classes_id):
-            min_prob = self.classes_probability[self.classes_id.index(class_id)]
-
-            if(class_probability >= min_prob):
-                if(self.recording == False):
-                    print("Detected. Start recording.")
-                    self.startRecordingPipeline()
-                else:
-                    self.start_recording_time = time.time()
-        else:
-            if(self.recording):
-                self.stop_recording_time = time.time()
-                diff = self.stop_recording_time - self.start_recording_time
-
-                if(diff >= self.min_recording_time_seconds):
-                    print("No detection. Stop recording.")
-                    self.stopRecordingPipeline()
+        for i in range(0,len(classes)):
+            class_id = classes[i] # Model class ID required to start recording
+            class_probability = probabilities[i] # Min probability needed to start recording
+
+            if(class_id in self.classes_id):
+                min_prob = self.classes_probability[self.classes_id.index(class_id)]
+
+                if(class_probability >= min_prob):
+                    if(self.recording == False):
+                        print("Detected. Start recording.")
+                        self.startRecordingPipeline()
+                    else:
+                        self.start_recording_time = time.time()
+            else:
+                if(self.recording):
+                    self.stop_recording_time = time.time()
+                    diff = self.stop_recording_time - self.start_recording_time
+
+                    if(diff >= self.min_recording_time_seconds):
+                        print("No detection. Stop recording.")
+                        self.stopRecordingPipeline()
```

Comment:
I believe this logic is flawed. The recording will only stop if there is a new prediction. But in the case were a person walks out of the scene and no new predictions are made, the recording will continue forever. We need a timeout mechanism that is restarted with every new valid detection.